File: docs/test-beds/go/fixture.go
// Go test bed
package main

import "fmt"

type Box struct {
	Value string
}

type Container struct {
	Items []Box
}

func (c *Container) Add(item Box) {
	c.Items = append(c.Items, item)
}

func (c *Container) Size() int {
	return len(c.Items)
}

func Process(input Box) *Container {
	c := &Container{}
	c.Add(input)
	fmt.Println(c)
	return c
}

const FOO int = 42

File: docs/test-beds/r/fixture.r
# R test bed
source("helper.R")

Box <- function(value) {
  list(value = value)
}

Container <- setRefClass("Container",
  fields = list(items = "list"),
  methods = list(
    add = function(item) { items <<- c(items, list(item)) },
    size = function() { length(items) }
  )
)

process <- function(input) {
  c <- Container$new(items = list())
  c$add(input)
  Helper.log(c)
  c
}

FOO <- 42
